src/Paket.Core/Dependencies/DependenciesFileParser.fs

```diff
@@ -25,29 +26,36 @@ module DependenciesFileParser =
         | PaketStrategy -> Some ResolverStrategy.Max
         | NoStrategy -> None
 
-    let twiddle (minimum:string) =
-        let promote index (values:string array) =
-            let parsed, number = Int32.TryParse values.[index]
-            if parsed then values.[index] <- (number + 1).ToString()
-            if values.Length > 1 then values.[values.Length - 1] <- "0"
-            values
-
-        let parts = minimum.Split '.'
-        let penultimateItem = Math.Max(parts.Length - 2, 0)
-        let promoted = parts |> promote penultimateItem
-        String.Join(".", promoted)
+    let twiddle (minimum:SemVerInfo) =
```

Comment:
changed to utilize existing SemVer parser, to handle multi segment prerelease ranges correctly
original code changed the last numeric segment and ended up with invalid upper constraint